lib/tasks/db.rake

```diff
@@ -1,49 +1,84 @@
 namespace :db do
 
-  desc 'Dumps the database to file'
-  task :dump => :environment do
+  desc 'Uploads a scrubbed database dump to S3'
+  task :'s3:upload', [:key] => :environment do |t, args|
+    args.with_defaults(:key => "db_#{ENV['APP_DOMAIN']}.sql")
 
-    url = ActiveRecord::Base.connection_config[:url]
-    host = ActiveRecord::Base.connection_config[:host]
-    db = ActiveRecord::Base.connection_config[:database]
+    Tempfile.open(['dump', '.sql']) do |dump_file|
+      puts "Dumping database to #{dump_file.path}..."
+      cmd = "pg_dump --no-owner --no-acl -v -c -f #{dump_file.path} --no-password #{connection_info}"
+      unless system cmd
+        raise 'Database dump failed'
+      end
 
-    cmd = "pg_dump --no-owner --no-acl -v -c -f #{dump_file} --no-password "
-    # Adapt the app database config for pg_dump. Use the db connection url, or
-    # fallback to using a local unix socket.
-    cmd += url || "--host #{host} --dbname #{db}"
+      Tempfile.open(['scrub', '.sql']) do |scrub_file|
+        puts 'Scrubbing...'
+        unless system "#{Rails.root}/bin/scrub.rb #{dump_file.path} > #{scrub_file.path}"
+          raise 'Database scrub failed'
+        end
 
-    puts 'Running pg_dump...'
-    unless system cmd
-      raise 'Database dump failed'
+        puts "Uploading to S3 as '#{args.key}'..."
+        s3 = Aws::S3::Resource.new
+        s3.bucket(s3_bucket).object(args.key).upload_file(scrub_file.path)
+        puts 'Upload complete'
+      end
     end
   end
 
-  desc 'Scrubs database dump file and uploads to S3'
-  task :s3upload => :dump do
-    begin
-      scrubbed_file = File.join(Dir.tmpdir, 'db_scrubbed.sql')
-      s3_key = "db_#{ENV['APP_DOMAIN']}.sql"
-      s3_bucket = ENV['AWS_DB_S3UPLOAD_BUCKET']
+  desc 'Import a database dump from S3'
+  task :'s3:import', [:key] => :environment do |t, args|
+    unless args.key.present?
+      abort 'You must specify the S3 key to import e.g. rake db:download[db_gov-au-beta.example.gov.au.sql]'
+    end
+
+    Rake::Task['db:drop'].invoke
+    Rake::Task['db:create'].invoke
```

Comment:
It might be a good idea to drop after downloading the database incase the download fails.